app/assets/javascripts/govuk_publishing_components/analytics-ga4/gtm-click-tracking.js

```diff
@@ -25,11 +25,25 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
           link_url: window.location.href.substring(window.location.origin.length),
           ui: JSON.parse(target.getAttribute('data-gtm-attributes')) || {}
         }
-        var ariaExpanded = this.checkExpandedState(target)
-        if (ariaExpanded) {
-          data.ui.state = ariaExpanded === 'false' ? 'opened' : 'closed'
+        var ariaExpanded = this.getClosestAttribute(target, 'aria-expanded')
+        /*
+          the details component uses an 'open' attribute instead of aria-expanded, so we need to check if we're on a details component.
+          since details deletes the 'open' attribute when closed, we need this boolean, otherwise every element which
+          doesn't contain an 'open' attr would be pushed to gtm as a closed element.
+        */
+        var canHaveOpenAttribute = target.getAttribute('data-module') ? target.getAttribute('data-module').includes('govuk-details') : false
+
+        if (ariaExpanded || canHaveOpenAttribute) {
           data.ui.text = data.ui.text || target.innerText
+
+          if (ariaExpanded) {
+            data.ui.state = (ariaExpanded === 'false') ? 'opened' : 'closed'
+          } else if (canHaveOpenAttribute) {
+            var openAttribute = this.getClosestAttribute(target, 'open')
```

Comment:
I see why, but this code now calls `getClosestAttribute` twice, which seems inefficient. It might be possible as part of altering the details component detection on line 34 to keep a hold of the `details` element itself as a variable, which could be referenced here to find the `open` attribute, instead of calling this function a second time?